daemons/fenced/fenced_commands.c

```diff
@@ -916,6 +933,10 @@ build_device_from_xml(xmlNode * msg)
         read_action_metadata(device);
         stonith__device_parameter_flags(&(device->flags), device->id,
                                         device->agent_metadata);
+    } else {
+        crm_trace("Adding timer to retry getting metadata.");
+        mainloop_timer_add("get_agent_metadata", 120 * 1000,
+                           TRUE, get_agent_metadata_cb, device);
```

Comment:
Despite its name, mainloop_timer_add() is a create function -- you have to call mainloop_timer_start() to use it, and  mainloop_timer_stop() when necessary. We'll have to track it and free it with mainloop_timer_del(); stonith_device_t is private API in daemons/fenced/pacemaker-fenced.h, so we can add a pointer there and free it in free_device()

I'm thinking maybe 30 seconds, any problems getting meta-data are generally very short-lived